root/static/scripts/release-editor/validation.js

```diff
@@ -123,13 +125,36 @@ $(function () {
 });
 
 
+// Search releases with the same barcode
+function searchExistingBarcode(field, barcode) {
+    request({
+        url: `/ws/2/release/?query=barcode:${barcode}&fmt=json`
+    }).done(data => {
+        if (data.releases.length) {
+            var msg = l(
+                `There already exists at least one release with that barcode.
+                 Please check on {url|this search page} whether the release you
+                 are entering is identical.`
+            );
+            var query = `barcode%3A${barcode}&type=release&method=advanced`;
+            field.existing(
+                expand2html(
+                    msg,
+                    { url: { href: `/search?query=${query}`, target: "_blank" }}
+                )
+            );
```

Comment:
It would make sense to me to just list the releases here. It shouldn't make it too big, since most barcodes aren't reused in many releases, and it would save a lot of time and make it much more likely for the user to check them. I think you should be able to just do a map on the results array and use DescriptiveLink to show the releases?

Reply:
I will test that idea and post a screenshot

Reply:
```suggestion
function searchExistingBarcode(field, barcode) {
    utils.search('release', `barcode:${barcode}`, 1).done(data => {
        if (data.releases.length) {
            const msg = l(
                `The following releases with that barcode are already in the
                 MusicBrainz database. Please make sure you are not adding an
                 exact duplicate of any of these:`,
            );
            const releaseList = ReactDOMServer.renderToString(
                <ul>
                    {data.releases.map(release => {
                        const cleanedRelease = new MB.entity.Release(
                            utils.cleanWebServiceData(release),
                        );
                        return (
                            <li key={release.id}>
                                <DescriptiveLink
                                    entity={cleanedRelease}
                                />
                            </li>
                        );
                    })}
                </ul>,
            );
            field.existing(msg + releaseList);
        }
    });
}
```

Reply:
Looks good. I added the relevant imports and pushed the commit. I didn't know about most of those helper functions :)